### ISSUE TITLE
OJS installDefaultNavigationMenus fairly slow in multijournal installations

### ISSUE THREAD

**Issue by ajnyga** [pkp/pkp-lib#3291]
Hi,

This is not a big issue, but I will leave it here anyway for your consideration.

While testing the upgrade from 3.0.2 to 3.1.0.1 I noticed that installDefaultNavigationMenus function here https://github.com/pkp/ojs/blob/master/dbscripts/xml/upgrade.xml#L83 takes a long time to finish. Around 5 minutes. 

If that would be something that would happen only when upgrading to a higher version, it would be ok. But I noticed that it is run everytime you use php tools/upgrade.php upgrade even if the version does not change. 

Maybe the installDefaultNavigationMenus should be moved somewhere where it is only triggered when the version actually changes. Unless there is something there that needs to be run every time you use the upgrade script. But in that case, is there maybe a way to make it faster. I am thinking about situations where you are for example installing plugins etc.

**Comment by asmecher** [comment]
@ajnyga, can you add a few quick metrics on your install? How many journals, and how many articles?

**Comment by ajnyga** [comment]
we have 78 journals and 35 200 submissions.

I just finished upgrading our production server to 3.1.0.1.
After the first upgrade run the size of the `navigation_menu_item_settings` was 442,368
I then moved our own plugins to the plugin folder and run the upgrade again to make sure the plugins are registered properly. After that second run the size of `navigation_menu_item_settings` is 327,680.  The other related tables are the same size.

But I do not understand why during the second run the table becomes smaller? I basically did not change anything in our install between those two runs, so it would be logical that the script would do nothing during the second run?
